### DISCUSSION TITLE
Branch off when feature freeze

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#7854] [General Discussions]
I suggest branching into `4.2` immediately when we feature freeze.

* Patches during the beta & rc stages are explicitly cherry-picked.
* For Pending PRs that should target later versions
    * Authors won't feel stucked for a month. 
    * We can still merge if it's really needed, but only into `master`.

**Comment by AThousandShips** [comment]
This will split focus and effort and complicate the process IMO, I don't see that PRs would be merged any faster during that period as focus is and should be on fixing and testing rather than new features

**Reply by timothyqiu** [reply]
This only suggests branching early. It doesn't mean we should split focus & effort to handle 4.3 PRs.

Members would always follow the requirement of feature freeze, regardless the branching policy.

This more for non-member contributors that focus on their own PRs. Freezing `master` directly means "your PR can't be merged currently because we're in feature freeze". Branching off and focusing `4.2` means "we are busy handling 4.2 specific PRs, so we could be slow to respond to non-4.2 PRs".

**Reply by AThousandShips** [reply]
Feature freeze doesn't stop anyone from working on thier own PRs so I don't see how this would change anything, these PRs won't be reviewed or approved or merged anyway

**Reply by timothyqiu** [reply]
If I were a random contributor and I'm told that my PR won't be reviewed or merged for a month or so, I would give up creating a PR, or leave my created PR there for a month and forget.

Say, during feature freeze, even if no bugs are reported and many members are in idle state, they can't review & merge these PRs just because anything included in `master` would be in `4.2`. It's a restriction of the branching policy. Currently, everyone _has to work on_ 4.2, instead of everyone is _focused on_ 4.2.

**Reply by AThousandShips** [reply]
But that's not the only reason we have feature freeze? The focus should be on bugs, not new features, so no PRs would be merged anyway and shouldn't be, so all it would add is extra work because all the work being done between feature freeze and release would have to be cherry picked, which isn't a zero effort process

The reason for feature freeze isn't just to keep unstable code out of the release branch, but to focus effort, splitting focus between the release cycle and future features would worsen both processes

Despite the strong suggestions to focus on bugs nothing is stopping anyone from reviewing feature PRs, but the post feature freeze cycle is hectic enough as it is without dividing work

**Reply by timothyqiu** [reply]
Being _able to_ merge 4.3 PRs doesn't necessarily mean one should _focus on_ merging 4.3 PRs.

We focus on 4.x development, but a few contributors do contribute to 3.x development regularly. This focus is not achieved by locking `3.x` branch.

**Comment by Zireael07** [comment]
Honestly, both ways of branching have their merits. It's hard to say when to branch off.

Also: I can't see a good way to avoid the problem that Timothy mentions, of PRs sitting unreviewed for a month or two because of feature freezes
